src/Microsoft.Azure.SignalR.Common/Exceptions/AzureSignalRUnauthorizedException.cs

```diff
@@ -9,12 +9,16 @@ namespace Microsoft.Azure.SignalR.Common
     [Serializable]
     public class AzureSignalRUnauthorizedException : AzureSignalRException
     {
-        private const string ErrorMessage = "Authorization failed. Make sure you provide the correct connection string and have access to the resource.";
+        private const string ErrorMessage = "Authorization failed. Please check the connection string and role assignments. The role assignments will take up to 30 minutes to take effect if it was added recently.";
```

Comment:
use different error messages for accessKey auth and aad auth?

Reply:
This message is only for AuthType = AzureAd

Reply:
The exception class name looks much more general though.

Reply:
It's a public class, it doesn't make sense if we raised an exception with its type name contains "AzureAd".